File: Source/Aladdin.CAPI.ANSI.PKCS11.Java/src/aladdin/capi/ansi/pkcs11/mac/HMAC_SHA2_512_256.java
package aladdin.capi.ansi.pkcs11.mac;
import aladdin.capi.pkcs11.*; 
import aladdin.pkcs11.*;
import java.io.*; 

///////////////////////////////////////////////////////////////////////////////
// Алгоритм вычисления имитовставки HMAC SHA2-512/256
///////////////////////////////////////////////////////////////////////////////
public class HMAC_SHA2_512_256 extends HMAC
{
	// конструктор
	public HMAC_SHA2_512_256(Applet applet) throws IOException { this(applet, 32); }

	// конструктор
	public HMAC_SHA2_512_256(Applet applet, int macSize) throws IOException 
    { 
        // сохранить переданные параметры
        super(applet, API.CKM_SHA512_256_HMAC, API.CKM_SHA512_256, macSize); 
    } 
}

File: Source/Aladdin.CAPI.ANSI.PKCS11/MAC/HMAC_SHA2_512_256.cs
using System;
using Aladdin.PKCS11; 

namespace Aladdin.CAPI.ANSI.PKCS11.MAC
{
	///////////////////////////////////////////////////////////////////////////////
	// Алгоритм вычисления имитовставки HMAC SHA2-512/256
	///////////////////////////////////////////////////////////////////////////////
	public class HMAC_SHA2_512_256 : HMAC
	{
		// конструктор
        public HMAC_SHA2_512_256(CAPI.PKCS11.Applet applet) : this(applet, 32) {}

		// конструктор
		public HMAC_SHA2_512_256(CAPI.PKCS11.Applet applet, int macSize) 
            
            // сохранить переданные параметры
            : base(applet, API.CKM_SHA512_256_HMAC, API.CKM_SHA512_256, macSize) {}
    }
}
